Metric Webapp

RileyManda opened the pull request:
I have completed the task of building the **Metrics web app** according to the stipulated [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md):

**The following implementations were made:** 🌱 

- [x] Created a single-page application (SPA) built with React and Redux.
- [x] Connected to an [API](https://docs.openaq.org/) to display metric data on the home page
- [x] Implemented the Home component to display a list of items from the API filtered by Air quality parameters.
- [x] Implemented detail view to display more details about a specific metric index.
- [x] Selected one main color for the Ui theme of the application
- [x] Implemented search filter to Home and Detail Component
- [x] Implemented a navigation button in the header to enable users to navigate to the home page.
- [x] Implemented redux and redux store using redux-toolkit. 
- [x] Create integration tests for the application using the React Testing Library
- [x] Setup redux store to ensure the data retrieved from the API is be stored in the Redux store
- [x] Hosted the application and included the link in readme
- [x] Recorded a presentation video for the  project and included the link in readme
- [x] Created unit tests for pure functions in the Redux code.
- [x] Created integration tests using react testing library and mocked the components.
- [x] Recorded presentation video showcasing the file structure,objective,and special features.


In addition, I have ensured that this project fulfills the following default requirements :

✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).


Please review my implementation and suggest any improvements and changes.

Thank you 🙏

TedLivist commented on README.md:
```diff
@@ -1 +1,281 @@
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to express my sincere gratitude to [Microverse](https://github.com/microverseinc), the dedicated reviewers, and collaborators. Your unwavering support, feedback, and collaborative efforts have played an immense role in making this journey a resounding success. I am truly grateful for your contributions and for being an integral part of my achievements. Thank you for your continued su ...
+
```
- [ ] Your acknowledgment section should contain credits for Nelson Sakwa whose design was used for this project, in keeping with the Creative Commons License

TedLivist commented on src/components/Home.js:
```diff
@@ -0,0 +1,137 @@
+
+  return (
+    <div>
+      <StyledHome>
+        <TopCard
+          backgroundImage={Map1}
+          location="USA"
+          views="890 Views"
+          footerText="Stats by air preassure"
+          toggleSearch={toggleSearch}
+          isSearchVisible={isSearchVisible}
+          setSearchKeyword={setSearchKeyword}
+        />
```
- [ ] Kindly modify the intro section of your app to provide more relevant information about the project you are building. Adding an arbitrary 860 views from the USA does not fit the project you are building. You are concentrating on Air pressure, then kindly provide information related to it.

  The intro section should provide more relevant information about the project, just like the design

  ![image](https://github.com/RileyManda/air-quality/assets/36542986/0a8c0643-e252-40a1-9b81-7e745895fb61)

TedLivist commented on src/components/Details.js:
```diff
@@ -0,0 +1,70 @@
+    }
+  };
+  return (
+    <div>
+      <TopCard
+        backgroundImage={Map4}
+        location={locationData.location}
+        views="700 Views"
+        footerText="CITY/TOWN BREAKDOWN-2013"
+        setSearchKeyword={setSearchKeyword}
+        toggleSearch={toggleSearch}
+        isSearchVisible={isSearchVisible}
+      />
```
- [ ] Kindly remove the unnecessary "700 views" text on the views on the details page, or provide more relevant information.

  ![image](https://github.com/RileyManda/air-quality/assets/36542986/137ff7c3-079f-4a8f-99fc-a2a766a880b3)

TedLivist commented on src/components/Home.js:
```diff
@@ -0,0 +1,137 @@
+                    <Link to={`/details/${location.location}/${uuidv4()}`} style={{ textDecoration: 'none' }}>
+                      <Card className={`content-card flex-container flex-column bold no-border flex-end ${index % 2 === 1 ? 'darker' : ''}`} style={{ backgroundImage: `url(${Map2})` }}>
+                        <Card.Header className="position-absolute top-0 end-0">
+                          <FontAwesomeIcon icon={faCircleRight} style={{ color: '#fff' }} />
+                        </Card.Header>
+                        <Card.Body
+                          style={{ display: 'flex', flexDirection: 'column', fluid: true }}
+                          className="card-content flex-container flex-column"
+                        >
+
+                          <div className="card-info-container flex-container flex-column">
+                            <Card.Title style={{ width: '6rem' }} className="title-text white-text bold ellipsis- multiline-2 text-wrap">
+                              {' '}
+                              {truncatedTitle.toUpperCase()}
+                            </Card.Title>
+                          </div>
+                          <div className="measurements-text white-text wrap-break">
+                            {location.measurements.map((metric) => {
+                              if (airQualityParameters.includes(metric.parameter)) {
+                                return (
+                                  <div key={measurementKey} className="measurements-item">
+                                    {metric.parameter}
+                                    {' '}
+                                    :
+                                    {' '}
+                                    {metric.value}
+                                    {' '}
+                                  </div>
+                                );
+                              }
+                              return null;
+                            })}
+                          </div>
+
+                        </Card.Body>
+                      </Card>
+                    </Link>
+                  </Col>
+                );
+              })
```
- [ ] The design for the items on the Homepage should follow the alternating color patterns as we have on the design. All the darker colors are on one side, but the design requires that you alternate the darker and lighter colors. Kindly do that.

  Your design without alternating color pattern | Required Design with alternating color pattern
  :---:|:---:
  ![image](https://github.com/RileyManda/air-quality/assets/36542986/c4cc4731-0f44-4efe-9753-1ea3fd3bb63d) | ![uiww](https://github.com/RileyManda/air-quality/assets/36542986/9e8c17f4-7aea-4732-a3c6-77c4daaa4354)

TedLivist commented on src/App.js:
```diff
@@ -0,0 +1,25 @@
+        <Header />
+        <Routes>
+          <Route path="/" element={<Home />} />
+          <Route path="details/:location" element={<Details />}>
```
- [ ] Each item on the home page should have a unique route. Right now, all your items have the same route: `/details`, but they should be something like so: `/details/:id`. Kindly fix that.

  For example (you don't need to use the same word), instead of all your routes having the same route like so:

  ```
  /details
  ```
  They can have this:
  ```
  /details/chipper-hill

  /details/san-antonio
  ```

RileyManda replied:
Thank you for the review and feedback @TedLivist 
I will make sure to make the requested changes and submit fo another review. 🙏

TedLivist reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @RileyManda,

Good job so far!
✔️ Nice use of color stripe in details page
✔️ Implementation of search functionality
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda commented:
I have completed the task of implementing the suggested changes as requested by the reviewer.

The following implementations were made: 🌱

- [x] Attributed Nelson Sakwa to the thank you section in the readme file.
- [x] Removed the useEffect hook from home and implemented the exact route of the details component
- [x]Implemented a checker pattern to the cards to match the original Ui design layout.

In addition, I have ensured that this project fulfills the following default requirements :

✅ - There are [no linter errors](https://github.com/microverseinc/linters-config).
✅ - Used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ - Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
✅ - Followed the list of the [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md) and  [JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).


Please review my implementation and suggest any improvements and changes.

Thank you 🙏

Ibizugbe reviewed (approved):
Hi @RileyManda 👋🏼👋🏼👋🏼, 

![approved](https://github.com/RileyManda/air-quality/assets/64643995/448738e7-f587-4178-9801-1a5f4898584f)


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] Acknnowledgement has been given to the author of the design as requested👍🏼.
- [x] Relevant information has been provided on the intro page as requested👍🏼.
- [x] The alternating color switch pattern has been implemented as requested👍🏼.
- [x] The routing of the page has been properly implemented as requested👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
